### PR TITLE
bsp: Add support for ESP-BOX

### PR DESCRIPTION
**Pull request by tore-espressif**
TODOs:

- [x] Touch driver
- [ ] IMU driver
- [ ] ADC driver
- [x] Mute button

Closes #21

### CODE REVIEW

**Comment by tore-espressif** [comment]
For initial version 0.0.1 there following todos:
* add ADC driver
* carve-out touch driver to separate component

@espzav PTAL

**Comment by espzav** [comment]
I tested this code with USB demo and it working right (with esp_lcd_touch). Some minor updates and questions added.

**Comment by tore-espressif** [comment]
Thanks @espzav , I cherry-picked your commit to this fork. I can't see the questions though...

File: esp-box/Kconfig [hunk 1]
```diff
@@ -0,0 +1,50 @@
+menu "Board Support Package"
+    config BSP_I2C_NUM
+        int "I2C peripheral index"
+        default 1
+        range 0 1
+        help
+            ESP32S3 has two I2C peripherals, pick the one you want to use.
+
+    config BSP_I2C_FAST_MODE
```
[the remark this answers is not in the document]
Comment (tore-espressif) [reply esp-box/Kconfig:9]: `BSP_I2C_CLK_SPEED_HZ` is derived from symbol which is then used during I2C initialization (esp_box.c line 35)

**Comment by tore-espressif** [comment]
@espzav All updated. Let's wait for esp_lcd_touch release and then we'll get back to this

File: esp-box/esp-box.c [hunk 2]
```diff
@@ -209,7 +209,7 @@ static void lvgl_port_display_init(void)
     esp_lcd_panel_reset(panel_handle);
     esp_lcd_panel_init(panel_handle);
     esp_lcd_panel_mirror(panel_handle, true, true);
-    esp_lcd_panel_disp_on_off(panel_handle, true);
+    esp_lcd_panel_disp_off(panel_handle, false); // In IDF v5.0 display is not turned on automatically
```
Comment (espzav) [inline esp-box/esp-box.c:212]: I would like see here more explaining comment e.g.:
/*
In IDF v5.0 should be called esp_lcd_panel_disp_on_off function for turn on display (display is not turned on automatically in IDF v5.0), but there is used deprecated function esp_lcd_panel_disp_off with the same behavior for compatibility with IDF v4.4
*/

Or something like that.

**Review by espzav — APPROVED** [review]

File: esp-box/esp-box.c (hunk 2 above)
Comment (tore-espressif) [reply esp-box/esp-box.c]: Thanks, comment updated. All commits squashed.

**Pull request merged**